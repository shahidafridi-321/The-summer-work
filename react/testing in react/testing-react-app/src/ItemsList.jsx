import React from "react";

export const ItemsList = () => {
	return (
		<div>
			<ul className="animals">
				<li>Cat</li>
				<li>Whale</li>
				<li>Lion</li>
				<li>Elephant</li>
				<li>Rhino</li>
			</ul>
		</div>
	);
};
