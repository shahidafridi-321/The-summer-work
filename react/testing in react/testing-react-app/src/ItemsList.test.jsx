import React from "react";
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { ItemsList } from "./ItemsList";

describe("App component", () => {
	it("list contains 5 animals", () => {
		// Render the App component
		render(<ItemsList />);
		screen.debug();

		// Locate the list and list items
		const listElement = screen.getByRole("list");
		const listItems = screen.getAllByRole("listitem");

		// Assert that the list is in the document
		expect(listElement).toBeInTheDocument();

		// Assert that the list has the correct class
		expect(listElement).toHaveClass("animals");

		// Assert that the list contains 5 items
		expect(listItems.length).toEqual(5);
	});
});
