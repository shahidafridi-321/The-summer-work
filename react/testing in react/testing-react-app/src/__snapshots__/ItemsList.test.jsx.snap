// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`List contains 5 animanls > Renders List  1`] = `
<div>
  <div>
    <ul
      class="animals"
    >
      <li>
        Cat
      </li>
      <li>
        Whale
      </li>
      <li>
        Lion
      </li>
      <li>
        Elephant
      </li>
      <li>
        Rhino
      </li>
    </ul>
  </div>
</div>
`;
